File: radar_areas.R
library(tidyverse)
library(sf)

areas <- read_sf("data/limite_de_bairros.geojson", crs = 4674) %>% 
  select(nome)

radar <- read_csv("data/ocr_radar.csv") %>% 
  st_as_sf(
    coords = c("camera_longitude", "camera_latitude"), 
    crs = 4674
  )

radar_areas <- st_intersection(radar, areas) %>% 
  mutate(
    lon = sf::st_coordinates(.)[,1],
    lat = sf::st_coordinates(.)[,2]
  ) %>% 
  st_drop_geometry()

write_csv(radar_areas, "data/radar_areas.csv")

File: radar_areas.py
import geopandas as gpd
import matplotlib.pyplot as plt

areas = gpd.read_file("data/limite_de_bairros.geojson")

areas.head()

areas.plot()
plt.show()


areas = areas[["nome", "geometry"]]

radar = pd.read_csv("data/ocr_radar.csv")

radar = gpd.GeoDataFrame(
  radar, 
  geometry = gpd.points_from_xy(radar.camera_longitude, radar.camera_latitude),
  crs = 4674
)

radar_areas = gpd.overlay(radar, areas, how = 'intersection')

radar_areas['lon'] = radar_areas.geometry.x
radar_areas['lat'] = radar_areas.geometry.y

radar_areas = radar_areas.drop(columns = 'geometry')
radar_areas.to_csv("data/radar_areas.csv", index = False)
